lib/spack/spack/spec.py

```diff
@@ -709,6 +804,11 @@ def from_node_dict(node):
         for name, enabled in node['variants'].items():
             spec.variants[name] = VariantSpec(name, enabled)
 
+        # XXX(deptypes): why are dependencies not meant to be read here?
```

Comment:
`from_node_dict` is used by `from_yaml`, which handles the deps -- it's just for one record.  Do you want to read them here for some reason?